src/lib/aws/timestream_manager.py

```diff
@@ -222,3 +238,60 @@ def get_MagneticStoreRetentionPeriodInDays(self):
         except Exception as err:
             error_message = f"Error getting MagneticStoreRetentionPeriodInDays for {self.db_name}.{self.table_name}: {err}."
             raise (TimestreamTableWriterException(error_message))
+
+
+    def get_earliest_writeable_time_for_table(self):
+        utc_tz = dateutil.tz.gettz('UTC')
+        return datetime.now(tz=utc_tz) - timedelta(hours=self.get_MemoryStoreRetentionPeriodInHours())
+
+class TimeStreamQueryRunner:
+    def __init__(self, timestream_query_client):
+        self.timestream_query_client = timestream_query_client
+
+    def execute_scalar_query(self, query):
+        """
+        Executes a scalar query (result = one row, one column) and returns the result.
+
+        Args:
+            query (str): The query to be executed.
+
+        Returns:
+            str: The result of the query.            
+        """
+        try:
+            response = self.timestream_query_client.query(QueryString=query)
+            first_record_data = response["Rows"][0]["Data"]
+            first_field_data = first_record_data[0]
+            if "ScalarValue" in first_field_data:
+                return first_field_data["ScalarValue"]
+            else:
+                return None
+        except Exception as e:
+            error_message = f"Error running query: {e}"
+            raise (TimestreamQueryException(error_message))
```

Comment:
I believe you do not need additional brackets here.

Reply:
fixed